scRNA-seq-advanced/setup/intro-diff-expression-setup.Rmd

```diff
@@ -0,0 +1,633 @@
+---
+title: 'Setup: Differential Expression Module'
+author: "Ally Hawkins for the Data Lab"
+date: "`r Sys.Date()`"
+output:
+  html_document:
+    toc: yes
+    toc_depth: '3'
+    df_print: paged
+  html_notebook:
+    toc: yes
+    toc_depth: 3
+    toc_float: yes
+    number_sections: no
+---
+
+
+## Setup and some initial exploration
+
+In this notebook we are planning to start with an object that has already been integrated. 
+This integrated object has a column that indicates the original batch (currently labeled `batch` but will be renamed to "sample") and a column that indicates cell types (`celltype`) in the `colData`.
+
+We can also grab any other additional sample-level metadata from the metadata file for this dataset. 
+We can use that to help identify groups of interest for differential expression. 
+
+In this first section, we explore the integrated object and identify libraries that would be good to use in setting up differential expression. 
+
+
+```{r}
+set.seed(2022)
+
+library(magrittr)
+library(SingleCellExperiment)
+library(ggplot2)
+```
+
+The merged and integrated datasets are available on S3 right now (`s3://sc-data-integration/scpca`), but after deciding what exactly we will use as input here, we will create the proper setup scripts and move the data to the server starting with the individual SCE objects used to create the integrated dataset. 
+
+```{r}
+# set up file paths and read in combined SCE and integrated SCE 
+data_dir <- file.path("..", "setup", "rms-integration")
+
+combined_sce_file <- file.path(data_dir, "dge", "SCPCP000005_merged_sce.rds")
+combined_sce <- readr::read_rds(combined_sce_file)
+
+
+integrated_sce_file <- file.path(data_dir, "dge", "SCPCP000005_integrated_fastmnn_sce.rds")
+integrated_sce <- readr::read_rds(integrated_sce_file)
+```
+
+```{r}
+# put the integrated PCA and UMAP into the combined object so all the data is together 
+reducedDim(combined_sce, "fastmnn_PCA") <- reducedDim(integrated_sce, "fastmnn_PCA")
+reducedDim(combined_sce, "fastmnn_UMAP") <- reducedDim(integrated_sce, "fastmnn_UMAP")
```

Comment:
I can change it in the one that's being added to the `combined_sce`, but the `integrated_sce` that I'm using we set to have `fastmnn` with lower case from the integration workflow. It will be changed when we use the setup snakemake workflow to create the object, but for now I'm going to leave it.